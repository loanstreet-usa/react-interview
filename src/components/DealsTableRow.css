.DealsTableRow {
  border-top: 1px solid #C1C3D1;
  border-bottom-: 1px solid #C1C3D1;
  color: #666B85;
  font-size: 16px;
  font-weight: normal;
  text-shadow: 0 1px 1px rgba(256, 256, 256, 0.1);
}

.DealsTableRow:hover td {
  background: #4E5066;
  color: #FFFFFF;
  border-top: 1px solid #22262e;
}

.DealsTableRow:first-child {
  border-top: none;
}

.DealsTableRow:last-child {
  border-bottom: none;
}

.DealsTable:nth-child(odd) .DealsTable--cell {
  background: #EBEBEB;
}

.DealsTable:nth-child(odd):hover .DeaslTable--cell {
  background: #4E5066;
}

.DealsTableRow:last-child .DealsTable--cell:first-child {
  border-bottom-left-radius: 3px;
}

.DealsTableRow:last-child .DealsTable--cell:last-child {
  border-bottom-right-radius: 3px;
}

.DealsTableRow--cell {
  background: #FFFFFF;
  padding: 20px;
  text-align: left;
  vertical-align: middle;
  font-weight: 300;
  font-size: 18px;
  text-shadow: -1px -1px 1px rgba(0, 0, 0, 0.1);
  border-right: 1px solid #C1C3D1;
}

.DealsTableRow--cell:last-child {
  border-right: 0px;
}

.removeDeal {
  color: white;
  padding: 5px;
  background-color: red;
  border-radius: 20px;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  box-shadow: 1px 1px 1px grey;
}

.publishDeal {
  color: white;
  float: right;
  background-color: green;
  box-shadow: 1px 1px 1px grey;
  padding: 5px 8px;
  border-radius: 20px;
  cursor: pointer;
}