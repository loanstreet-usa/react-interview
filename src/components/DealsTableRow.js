import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { deleteDeal, publishDeal } from '../actions';

import './DealsTableRow.css';

function currencyAmountToString(amount) {
  return "$" + amount.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
}

class DealsTableRow extends Component {
  static propTypes = {
    deal: PropTypes.shape({
      institution: PropTypes.string.isRequired,
      dealType: PropTypes.string.isRequired,
      dealSize: PropTypes.string.isRequired,
      isPublished: PropTypes.bool.isRequired
    }).isRequired
  }

  handleDeleteDeal = (id) => {
    this.props.deleteDeal(id);
  }

  render() {
    const { deal: { institution, dealType, dealSize, isPublished, id } } = this.props;
    return (
      <tr className="DealsTableRow">
        <td className="DealsTableRow--cell">{institution}</td>
        <td className="DealsTableRow--cell">{dealType}</td>
        <td className="DealsTableRow--cell">{currencyAmountToString(dealSize)}</td>
        <td className="DealsTableRow--cell">{isPublished ? 'Yes' :
          <div>
            No
            <span className='publishDeal' onClick={() => this.props.handlePublishDeal(id)}>Publish</span>
          </div>}</td>
        <td className="DealsTableRow--cell"><div className='removeDeal' onClick={() => this.handleDeleteDeal(id)}>x</div></td>
      </tr>
    )
  }
}

export default connect(null, { deleteDeal, publishDeal })(DealsTableRow);
