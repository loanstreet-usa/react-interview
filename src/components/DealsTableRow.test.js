import React from 'react';
import { create } from 'react-test-renderer';
import DealsTableRow from './DealsTableRow';
it('Renders', () => {
    const dealMock = {
        id: 1,
        institution: 'LS Credit Union',
        dealSize: '1000000',
        dealType: 'Consumer Auto',
        isPublished: true,
    };
    const target = create(<DealsTableRow
        deal={dealMock}
        onRemoveDeal={jest.fn()} />);

    expect(target).toMatchSnapshot();

});