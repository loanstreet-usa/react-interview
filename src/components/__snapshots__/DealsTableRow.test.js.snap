// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
<tr
  className="DealsTableRow"
>
  <td
    className="DealsTableRow--cell"
  >
    LS Credit Union
  </td>
  <td
    className="DealsTableRow--cell"
  >
    Consumer Auto
  </td>
  <td
    className="DealsTableRow--cell"
  >
    $1,000,000
  </td>
  <td
    className="DealsTableRow--cell"
  >
    Yes
  </td>
</tr>
`;
